[yyueyii] iP

yyueyii opened the pull request:
# Mario
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Mario frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~fast~~ SUPER FAST TO USE

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)
2. double click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**

Features:
- [ ] Managing tasks
- [ ] Managing deadlines

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level 0 completed

Commit:
Compelted Level-1

Commit:
Completed Level-2

Commit:
Completed Level-3

Commit:
Completed Level-4

Commit:
Added error handling, completed Level-5

Commit:
Completed Level-5, error handling

Commit:
Completed Level-6, added delete function

Commit:
Added enums

Commit:
Modify code, add Parser and Ui classes

Commit:
Add Storage class. Completed Level-7

Commit:
Add Storage class, complete Level-7

yyueyii committed:
Merge pull request #1 from yyueyii/branch-level-7

Branch level 7

Commit:
Add Dates and Times, complete Level-8

yyueyii committed:
Merge pull request #2 from yyueyii/branch-Level-8

Add Dates and Times, complete Level-8

Commit:
Complete A-MoreOOP

Commit:
Add packages

Commit:
Merge branch 'master' of https://github.com/yyueyii/ip

* 'master' of https://github.com/yyueyii/ip:

yyueyii committed:
Merge pull request #3 from yyueyii/A-Packages

Add packages

Commit:
Add Tasks package

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

yyueyii committed:
Merge pull request #4 from yyueyii/add-gradle-support

Add gradle support

yyueyii committed:
Merge pull request #5 from yyueyii/A-Packages

Add Tasks package

Commit:
Merge branch 'master' of https://github.com/yyueyii/ip

* 'master' of https://github.com/yyueyii/ip:
  Add Tasks package
  Add packages

Commit:
Set up Gradle

Commit:
Add Ui Test and TaskList Test

Commit:
Create JAR file

Commit:
Add JavaDoc

Commit:
Change switch statement format

Commit:
Edit to fit coding standard

Commit:
Add find function. Complete Level-9.

Commit:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

* branch-A-JavaDoc:
  Add JavaDoc

Commit:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

* branch-A-CodingStandard:
  Edit to fit coding standard
  Change switch statement format

# Conflicts:
#	src/main/java/Mario/Mario.java
#	src/main/java/Mario/Parser.java

Vangmay commented on src/main/java/Mario/CommandType.java:
```diff
@@ -0,0 +1,37 @@
+
+    /**
+     * Returns the Enum command type of the command string.
+     *
```
Perhaps you can consider adding @params tag to this javadoc comment to provide more details.

Vangmay commented on src/main/java/Mario/CommandType.java:
```diff
@@ -0,0 +1,37 @@
+     * Returns the Enum command type of the command string.
+     *
+     */
+    public static CommandType fromString(String commandString) {
```
I think it might be a good idea to add `throws IllegalArgumentException` at the end of this method.

Vangmay commented on src/main/java/Mario/Parser.java:
```diff
@@ -0,0 +1,318 @@
+                    "Mamma mia! The event name cannot be empty!\n"));
+
+        } else {
+            String info;
```
Consider adding variable declarations at the top of the method, as it allows the programmer to know what variables are being used beforehand.

Vangmay commented on src/main/java/Mario/Parser.java:
```diff
@@ -0,0 +1,318 @@
+     * @param dateTimeString In the format yyyy-MM-dd HHmm.
+     * @return
+     */
+    public static LocalDateTime parseDateTime(String dateTimeString) {
```
I like that you have created a specific function to handle date-times so it is possible to change the date-time format easily :)

Vangmay commented on src/main/java/Mario/Tasks/ToDo.java:
```diff
@@ -0,0 +1,32 @@
+
+import Mario.Tasks.Task;
+
+public class ToDo extends Task {
```
Please consider renaming this class to `Todo` so that it follows the coding standards

Vangmay commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,88 @@
+import Mario.TaskList;
+import Mario.Tasks.Task;
+import org.junit.jupiter.api.Test;
```
I think you should add a space between line 2 and 3 to keep the imports organized and improve readability.

Vangmay commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,88 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Perhaps an alternative to this statement could be to import each and every Assertion you have used individually?

Vangmay reviewed (commented):
LGTM. Overall, I found the code easy to read except a few places where variable declarations were not given at the start of the methods. I also found some minor coding standard violations like the wildcard import and a minor nit regarding the naming of `ToDo` class.Besides that I think you have done a great job at creating a bot named Mario and captures the essence of the character.

vinc3leong commented on src/main/java/Mario/Ui.java:
```diff
@@ -0,0 +1,23 @@
+     * @param content
+     * @return
+     */
+    public String betweenLines(String content) {
```
Consider naming this function printSeparatorLine for more clarity

vinc3leong commented on src/main/java/Mario/Mario.java:
```diff
@@ -0,0 +1,232 @@
+                + "     Now you have " + lst.length() + " task(s) in your list!\n"));
+    }
+
+    public int getNumTasks() {
```
Perhaps naming the function getSize or size would make its purpose clearer to future reviewers

vinc3leong commented on src/main/java/Mario/Storage.java:
```diff
@@ -0,0 +1,84 @@
+import Mario.Tasks.Task;
+import Mario.Tasks.ToDo;
+
+import java.io.*;
```
Maybe consider using explicit imports instead of wildcards? It would make the code more readable and ensure only the necessary classes or methods are imported. :)

vinc3leong reviewed (commented):
I've left some suggestions that I think will help enhance the overall quality of the code. Overall great job! :))

Commit:
Add ./config/checkstyle files

Commit:
Configure checkstyle

Commit:
Add Main and Launcher classes

Commit:
Add images, DialogBox class

Commit:
Set up FXML

Commit:
Edit functions to handle GUI interactions

Commit:
Add assertions to MainWindow and Parser classes for internal validation

This commit enhances the robustness of the chat bot by adding assertions
to the MainWindow and Parser classes. The assertions prevent users from
entering blank inputs and ensures that the command names and formats
are valid.

Commit:
Improve code quality to adhere to SLAP, naming conventions, and formatting

Commit:
Improve code quality

Commit:
Modify task finding to use streams for improved readability

yyueyii committed:
Merge pull request #6 from yyueyii/branch-A-Streams

Modify task finding to use streams for improved readability

Commit:
Merge branch 'master' of https://github.com/yyueyii/ip

* 'master' of https://github.com/yyueyii/ip:
  Modify task finding to use streams for improved readability
  Improve code quality

# Conflicts:
#	src/main/java/Mario/Mario.java

Commit:
Add extension B - FixedDurationTasks

Commit:
Improve GUI

yyueyii committed:
Update README.md

Commit:
Add Ui.png

Commit:
Merge branch 'master' of https://github.com/yyueyii/ip

* 'master' of https://github.com/yyueyii/ip:
  Update README.md

yyueyii committed:
Update README.md

yyueyii committed:
Update README.md

yyueyii committed:
Update README.md

Commit:
Correct package naming

Commit:
Merge branches 'master' and 'master' of https://github.com/yyueyii/ip

* 'master' of https://github.com/yyueyii/ip:
  Update README.md
  Update README.md
  Update README.md

* 'master' of https://github.com/yyueyii/ip:
  Update README.md
  Update README.md
  Update README.md

Commit:
Fix Launcher issue

yyueyii committed:
Update README.md
